### ISSUE TITLE
[Arrow Optics] Generated optics code does not escape Kotlin keywords in package names

### ISSUE THREAD

**Issue by hrafnthor** [arrow-kt/arrow#2924]
#### Description:

Testing out using 'arrow-optics' in a project of mine, it immediatly caused compilation errors due to package names in the generated code containing unescaped Kotlin keywords from the originator package. 

For instance the following code is from my project
```kotlin
package `is`.hth.projectname

import arrow.optics.optics

@optics
public data class Audio(
    val url: String = "",
    val secureUrl: String = "",
    val type: String = ""
) {
    public companion object
}
```
And it generates the following:

```kotlin
package is.hth.projectname

public inline val is.hth.projectname.Audio.Companion.iso: arrow.optics.Iso<is.hth.projectname.Audio, kotlin.Triple<kotlin.String, kotlin.String, kotlin.String>> inline get() = arrow.optics.Iso(
  get = { audio: is.hth.projectname.Audio -> kotlin.Triple(audio.`url`, audio.`secureUrl`, audio.`type`) },
  reverseGet = { triple: kotlin.Triple<kotlin.String, kotlin.String, kotlin.String> -> is.hth.projectname.Audio(triple.first, triple.second, triple.third) }
)

```

In this case the keyword is being used in the package name to reference a reverse TLD.

A list of TLD's that match Kotlin's reserved keywords:
- as: American Samoa
- by: Belarus
- do: Dominican Republic
- in: India
- is: Iceland
- it: Italy

#### Versions
arrow: 1.1.5
kotlin: 1.8.0
ksp: 1.8.0-1.0.8

**Comment by nomisRev** [comment]
> If my understanding is correct, would a pull request looking at solving this issue be appreciated?

Hey @hrafnthor,
Thanks for opening an issue with details reproducible steps! A PR would absolutely by greatly appreciated 🙏

**Comment by hrafnthor** [comment]
I have staged a PR https://github.com/arrow-kt/arrow/pull/2925 for consideration. As this is my first contribution to the arrow project, all constructive feedback is appreciated as any norms and coding styles within it are currenlty unknown to me.

After going over the [contribution](https://github.com/arrow-kt/arrow/blob/main/CONTRIBUTING.md) document, I noticed that use of `spotless` is required for contribution PRs. However it does not seem that `spotless` is configured for the project. Or perhaps I am missunderstanding?

I did notice that all the unit tests contained in  `DSLTests.kt`, `IsoTests.kt``, `OptionalTests.kt`, `PrimsTests.kt` broke as soon they were given package declarations which contained Kotlin keywords. My local changes that remedy those issues were held back to keep the scope relevant to this issue.

**Comment by hrafnthor** [comment]
> I am not sure why those tests would break 

Sorry, I was probably not too clear on what I meant 🙂

As can be seen in the changes made in the PR [here](https://github.com/arrow-kt/arrow/pull/2925/commits/3035cdd0c4cf5b91af0cf3fb45977ce6945babde#diff-525a11f1c258ab4a716bcedb62be7a8420ffc7623fbf09efedf5f640d2fde38f) there were mock `package` definitions added to the test cases for Lens to make sure the sanitization changes work. 

If the same dummy package field is added to any of the other test cases in `DSLTests.kt`, `IsoTests.kt`, `OptionalTests.kt`, `PrimsTests.kt` then the generated code will be cause errors. 

> Feel free to add the fix to the PR, it sounds like the changes are related.

I've added those other changes to the PR now :+1:

**Comment by nomisRev** [comment]
Closed by #2925
